src/storage/mongodb/TransactionStorage.ts

```diff
@@ -445,6 +445,14 @@ export default class TransactionStorage {
     }
     // Create Aggregation
     const aggregation = [];
+    if (!Utils.isEmptyArray(params.visualTagIDs)) {
```

Comment:
This may kill the perfs but let's see...

Reply:
Yes, as discussed we can't put visualTagIDs in the transaction document since it can be changed.